src/main/java/luke/Luke.java

```diff
@@ -0,0 +1,301 @@
+            }
+            if (idx < 0 || idx >= tasks.size()) {
+                throw new IncorrectInput("Invalid index");
+            }
+            Task taskToDelete = tasks.get(idx);
+            tasks.remove(taskToDelete);
+            printReply(String.format("Removed task:\n  %s\nNow you have %d %s in the list.",
+                    taskToDelete.toString(), tasks.size(), tasks.size() > 1 ? "tasks" : "task"));
+            break;
+        default:
+            throw new InvalidCommand("Invalid command");
+        }
+    }
+
+    private static void sendMessage(String userInput) {
+        String[] inputArr = userInput.split(" ");
+        String command = inputArr[0];
+
+        if (command.equalsIgnoreCase("bye")) {
+            executeCommand(CommandType.BYE, inputArr);
+        } else if (command.equalsIgnoreCase("list")) {
+            executeCommand(CommandType.LIST, inputArr);
+        } else if (command.equalsIgnoreCase("mark")) {
+            try {
+                executeCommand(CommandType.MARK, inputArr);
+            } catch (LukeException e) {
+                printReply(e.getMessage());
+            }
+        } else if (command.equalsIgnoreCase("unmark")) {
+            try {
+                executeCommand(CommandType.UNMARK, inputArr);
+            } catch (LukeException e) {
+                printReply(e.getMessage());
+            }
+        } else if (command.equalsIgnoreCase("todo")){
+            try {
+                executeCommand(CommandType.TODO, inputArr);
+            } catch (InsufficientArguments e) {
+                printReply(e.getMessage());
+            }
+        } else if (command.equalsIgnoreCase("deadline")){
+            try {
+                executeCommand(CommandType.DEADLINE, inputArr);
+            } catch (InsufficientArguments e) {
+                printReply(e.getMessage());
+            }
+        } else if (command.equalsIgnoreCase("event")) {
+            try {
+                executeCommand(CommandType.EVENT, inputArr);
+            } catch (InsufficientArguments e) {
+                printReply(e.getMessage());
+            }
+        } else if (command.equalsIgnoreCase("delete")) {
+            try {
+                executeCommand(CommandType.DELETE, inputArr);
+            } catch (InsufficientArguments | IncorrectInput e) {
+                printReply(e.getMessage());
+            }
+        } else {
+            throw new InvalidCommand("Invalid command");
+        }
+    }
+
+    private static void printGreeting() {
+        printDivider();
+        System.out.println(LOGO);
+        System.out.println("Hi im Luke!\nWhat can I do for you? :)");
+        printDivider();
+    }
+
+    private static void loadSingleTask(String taskStr) {
+        String[] taskStrArr = taskStr.split("\\|");
+        switch (taskStrArr[TASK_TYPE_INDEX]) {
+        case "T" -> tasks.add(new ToDo(taskStrArr[DESCRIPTION_INDEX], taskStrArr[ISDONE_INDEX].equals("1")));
```

Comment:
To prevent error-prone code, perhaps you should use colons ':' instead of arrow "->" instead?
You can also use default branches.